tests/test_coin.py

```diff
@@ -0,0 +1,330 @@
+    )
+
+    c = Coin(parent_coin, puzzle_hash, 0xFFFFFF)
+    assert (
+        c.name()
+        == sha256(parent_coin + puzzle_hash + bytes([0, 0xFF, 0xFF, 0xFF])).digest()
+    )
+
+    c = Coin(parent_coin, puzzle_hash, 0xFFFFFFFF)
+    assert (
+        c.name()
+        == sha256(
+            parent_coin + puzzle_hash + bytes([0, 0xFF, 0xFF, 0xFF, 0xFF])
+        ).digest()
+    )
+
+    c = Coin(parent_coin, puzzle_hash, 0xFFFFFFFFFF)
+    assert (
+        c.name()
+        == sha256(
+            parent_coin + puzzle_hash + bytes([0, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF])
+        ).digest()
+    )
+
+    c = Coin(parent_coin, puzzle_hash, 0xFFFFFFFFFFFF)
+    assert (
+        c.name()
+        == sha256(
+            parent_coin + puzzle_hash + bytes([0, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF])
+        ).digest()
+    )
+
+    c = Coin(parent_coin, puzzle_hash, 0xFFFFFFFFFFFFFF)
+    assert (
+        c.name()
+        == sha256(
+            parent_coin
+            + puzzle_hash
+            + bytes([0, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF])
+        ).digest()
+    )
+
+    c = Coin(parent_coin, puzzle_hash, 0xFFFFFFFFFFFFFFFF)
+    assert (
+        c.name()
+        == sha256(
+            parent_coin
+            + puzzle_hash
+            + bytes([0, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF])
+        ).digest()
+    )
+
+    c = Coin(parent_coin, puzzle_hash, 0x7F)
+    assert c.name() == sha256(parent_coin + puzzle_hash + bytes([0x7F])).digest()
+
+    c = Coin(parent_coin, puzzle_hash, 0x7FFF)
+    assert c.name() == sha256(parent_coin + puzzle_hash + bytes([0x7F, 0xFF])).digest()
+
+    c = Coin(parent_coin, puzzle_hash, 0x7FFFFF)
+    assert (
+        c.name()
+        == sha256(parent_coin + puzzle_hash + bytes([0x7F, 0xFF, 0xFF])).digest()
+    )
+
+    c = Coin(parent_coin, puzzle_hash, 0x7FFFFFFF)
+    assert (
+        c.name()
+        == sha256(parent_coin + puzzle_hash + bytes([0x7F, 0xFF, 0xFF, 0xFF])).digest()
+    )
+
+    c = Coin(parent_coin, puzzle_hash, 0x7FFFFFFFFF)
+    assert (
+        c.name()
+        == sha256(
+            parent_coin + puzzle_hash + bytes([0x7F, 0xFF, 0xFF, 0xFF, 0xFF])
+        ).digest()
+    )
+
+    c = Coin(parent_coin, puzzle_hash, 0x7FFFFFFFFFFF)
+    assert (
+        c.name()
+        == sha256(
+            parent_coin + puzzle_hash + bytes([0x7F, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF])
+        ).digest()
+    )
+
+    c = Coin(parent_coin, puzzle_hash, 0x7FFFFFFFFFFFFF)
+    assert (
+        c.name()
+        == sha256(
+            parent_coin
+            + puzzle_hash
+            + bytes([0x7F, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF])
+        ).digest()
+    )
+    c = Coin(parent_coin, puzzle_hash, 0x7FFFFFFFFFFFFFFF)
+    assert (
+        c.name()
+        == sha256(
+            parent_coin
+            + puzzle_hash
+            + bytes([0x7F, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF])
+        ).digest()
+    )
```

Comment:
Maybe some `0x80` prefixes too? Since that's where the extra-byte wraparound happens.